### PR TITLE
Task 9

### PR DESCRIPTION
**Pull request by viacheslav-karnaukh**
Cart service: https://wed4qpuik0.execute-api.eu-west-1.amazonaws.com/dev/api/profile/cart
Product service: https://cc3mwqo2y1.execute-api.eu-west-1.amazonaws.com/dev/products

BFF service (/cart and /products to test corresponding services): http://viacheslav-karnaukh-bff-api-development.eu-west-1.elasticbeanstalk.com

FE app: https://d57iuemqb7f9r.cloudfront.net/admin/products

- [x] 3 - A working and correct express application should be in the bff-service folder.
- [x] 5 - The bff-service should be deployed with Elastic Beanstalk. The bff-service call should be redirected to the appropriate service. The response from the bff-service should be the same as if the recipient service was called directly.

- [x] +1 - Add a cache at the bff-service level for a request to the getProductsList function of the product-service. The cache should expire in 2 minutes.
How to test:
Get products list
Create new product
Get products list - result shouldn’t have new product
Wait more than 2 minutes
Get products list - result should have new product
- [x] +1 - Use NestJS to create bff-service instead of express

### CODE REVIEW

File: bff-service/.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+products=https://cc3mwqo2y1.execute-api.eu-west-1.amazonaws.com/dev
```
Comment (OHalahan) [inline bff-service/.env:1]: `.env` should not be present in repo directly

**Comment by Max-Korsakov** [comment]
If bff-service cannot find recipientURL by the {recipient-service-name}, return a 'Cannot process request’ doesn't implemented

image

**Review by OHalahan — APPROVED** [review]

File: bff-service/.env (hunk 1 above)
Comment (viacheslav-karnaukh) [reply bff-service/.env:1]: it was added intentionally for review purpose, the data is public. anyway, this functionality shouldn't be done using .env variables. this is just a task requirement

**Review by viacheslav-karnaukh — COMMENTED** [review]
> If bff-service cannot find recipientURL by the {recipient-service-name}, return a 'Cannot process request’ doesn't implemented
> 
> image

you are using incorrect link, BFF service (/cart and /products to test corresponding services): http://viacheslav-karnaukh-bff-api-development.eu-west-1.elasticbeanstalk.com

**Pull request merged**